File: backend/sdk/src/resources/webhooks.ts
import type { SendIXClient } from '../client.js'
import type { WebhookOptions, WebhookResponse } from '../types.js'

export class Webhooks {
  constructor(private readonly client: SendIXClient) {}

  /**
   * Register a new webhook endpoint for one or more events.
   * @param options - Target URL, event types, and signing secret.
   */
  async create(options: WebhookOptions): Promise<WebhookResponse> {
    return this.client.request<WebhookResponse>('POST', '/v1/webhooks', options)
  }

  /**
   * List all registered webhooks for the current account.
   */
  async list(): Promise<{ success: boolean; data?: { webhooks: any[] } }> {
    return this.client.request<{ success: boolean; data?: { webhooks: any[] } }>(
      'GET',
      '/v1/webhooks'
    )
  }

  /**
   * Delete a webhook by ID.
   * @param id - The webhook ID to remove.
   */
  async delete(id: string): Promise<{ success: boolean }> {
    return this.client.request<{ success: boolean }>('DELETE', `/v1/webhooks/${id}`)
  }

  /**
   * List recent webhook delivery attempts and their results.
   */
  async deliveries(): Promise<{ success: boolean; data?: { deliveries: any[] } }> {
    return this.client.request<{ success: boolean; data?: { deliveries: any[] } }>(
      'GET',
      '/v1/webhooks/deliveries'
    )
  }
}

File: backend/sdk-python/sendix/resources/webhooks.py
"""Webhook resource for the SendIX SDK."""

from ..client import SendIXClient
from ..types import WebhookOptions


class Webhooks:
    """Wraps the SendIX webhook API endpoints.

    Usage::

        from sendix import SendIX
        client = SendIX("sk_live_xxx")
        wh = client.webhooks.create({"url": "https://…", "events": […], "secret": "…"})

    Args:
        client: An authenticated :class:`sendix.client.SendIXClient` instance.
    """

    def __init__(self, client: SendIXClient):
        self._client = client

    def create(self, options: WebhookOptions) -> dict:
        """Register a new webhook endpoint.

        Args:
            options: Webhook configuration (see :class:`WebhookOptions`).

        Returns:
            The created webhook object with its ID and signing secret.
        """
        return self._client.request("POST", "/v1/webhooks", body=dict(options))

    def list(self) -> dict:
        """List all registered webhooks for the authenticated account.

        Returns:
            API response containing a list of webhook objects.
        """
        return self._client.request("GET", "/v1/webhooks")

    def delete(self, webhook_id: str) -> dict:
        """Remove a webhook by its ID.

        Args:
            webhook_id: UUID of the webhook to delete.

        Returns:
            Confirmation message from the API.
        """
        return self._client.request("DELETE", f"/v1/webhooks/{webhook_id}")

    def deliveries(self) -> dict:
        """Retrieve recent webhook delivery attempts.

        Returns:
            API response containing a list of delivery logs.
        """
        return self._client.request("GET", "/v1/webhooks/deliveries")
